servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncCloseables.java

```diff
@@ -57,27 +58,19 @@ public static Completable closeAsyncGracefully(AsyncCloseable closable, long tim
      * @return A new {@link ListenableAsyncCloseable}.
      */
     public static ListenableAsyncCloseable emptyAsyncCloseable() {
-        return new ListenableAsyncCloseable() {
-
-            private final CompletableProcessor onClose = new CompletableProcessor();
-            private final Completable closeAsync = new SubscribableCompletable() {
-                @Override
-                protected void handleSubscribe(final Subscriber subscriber) {
-                    onClose.onComplete();
-                    onClose.subscribeInternal(subscriber);
-                }
-            };
-
-            @Override
-            public Completable closeAsync() {
-                return closeAsync;
-            }
+        return new EmptyAsyncClosable(new CompletableProcessor());
+    }
 
-            @Override
-            public Completable onClose() {
-                return onClose;
-            }
-        };
+    /**
+     * Creates an empty {@link ListenableAsyncCloseable} that does nothing when
+     * {@link ListenableAsyncCloseable#closeAsync()} apart from completing the
+     * {@link ListenableAsyncCloseable#onClose()}.
+     * @param maxSubscribers The maximum amount of {@link CompletableSource.Subscriber} that can be subscribed to
+     * {@link ListenableAsyncCloseable#onClose()} on the return value.
+     * @return A new {@link ListenableAsyncCloseable}.
+     */
+    public static ListenableAsyncCloseable emptyAsyncCloseable(int maxSubscribers) {
```

Comment:
We or users may have special implementations of `CompletableSource.Processor`, consider taking the `Processor` instead of targeting specific features at this level.

Overall, I feel it will be hard for users to think about number of subscribers when they create a `ListenableAsyncCloseable`.